taggui/widgets/auto_captioner.py

```diff
@@ -410,6 +411,30 @@ def start_or_cancel_captioning(self):
             # Start captioning.
             self.generate_captions()
 
+    @Slot()
+    def notify_finished(self):
+        if not self.captioning_thread.is_canceled:
+            if self.captioning_thread.is_error:
+                sound = 'error'
+                icon = QMessageBox.Icon.Critical
+                text = ('Captioning error. See the Auto-Captioner console for '
+                        'more information.')
+            else:
+                sound = 'success'
+                icon = QMessageBox.Icon.Information
+                text = 'Captioning finished'
+            sound_effect = QSoundEffect()
+            sound_effect.setSource(QUrl.fromLocalFile(f'sounds/{sound}.wav'))
+            sound_effect.play()
+
+            dialog = QMessageBox()
+            dialog.setIcon(icon)
+            dialog.setText(text)
+            dialog.show()
+            dialog.raise_()
+            dialog.activateWindow()
```

Comment:
Are you sure these are necessary? Just leaving `dialog.exec()` results in the same behavior for me.